schema/snapcraft.json

```diff
@@ -0,0 +1,792 @@
+            "minitems": 1,
+            "uniqueItems": true,
+            "items": {
+                "oneOf": [
+                    {
+                        "type": "string",
+                        "usage": "<string>"
+                    },
+                    {
+                        "type": "object",
+                        "usage": "on <selector>[,<selector>...]:",
+                        "additionalProperties": false,
+                        "patternProperties": {
+                            "^on\\s+.+$": {
+                                "$ref": "#/definitions/grammar-array"
+                            }
+                        }
+                    },
+                    {
+                        "type": "object",
+                        "usage": "to <selector>[,<selector>...]:",
+                        "additionalProperties": false,
+                        "patternProperties": {
+                            "^to\\s+.+$": {
+                                "$ref": "#/definitions/grammar-array"
+                            }
+                        }
+                    },
+                    {
+                        "type": "object",
+                        "usage": "try:",
+                        "additionalProperties": false,
+                        "patternProperties": {
+                            "^try$": {
+                                "$ref": "#/definitions/grammar-array"
+                            }
+                        }
+                    },
+                    {
+                        "type": "object",
+                        "usage": "else:",
+                        "additionalProperties": false,
+                        "patternProperties": {
+                            "^else$": {
+                                "$ref": "#/definitions/grammar-array"
+                            }
+                        }
+                    }
+                ]
+            }
+        }
+    },
+    "title": "snapcraft schema",
+    "type": "object",
+    "properties": {
+        "build-packages": {
+            "$ref": "#/definitions/grammar-array",
+            "description": "top level build packages."
+        },
+        "adopt-info": {
+            "type": "string",
+            "description": "name of the part that provides source files that will be parsed to extract snap metadata information"
+        },
+        "name": {
+            "description": "name of the snap package",
+            "allOf": [
+                {
+                    "type": "string",
+                    "validation-failure": "snap names need to be strings.",
+                    "maxLength": 40
+                },
+                {
+                    "pattern": "^[a-z0-9-]*[a-z][a-z0-9-]*$",
+                    "validation-failure": "{.instance!r} is not a valid snap name. Snap names can only use ASCII\nlowercase letters, numbers, and hyphens, and must have at least one\nletter.\n"
```

Comment:
can you remove the `\n` from these `validation-failure` messages, the original yaml did not have them (they did not use `|` to denote verbatim entries).